### PR TITLE
Fixes ambiguity on two diff structs with same arg name

### PR DESCRIPTION
**Pull request by digorithm**
Fixes https://github.com/FuelLabs/fuels-rs/issues/77. 

Also simplifies some things related to custom types.

### CODE REVIEW

**Comment by digorithm** [comment]
@vnepveu just a heads up: this PR changes some things related to custom types, which means I had to make some small tweaks to the custom types abigen that you've created recently. No big changes though.

File: fuels-abigen-macro/tests/test_projects/two-structs/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+[package]
+name = "demo"
+version = "0.1.0"
+authors = ["Rodrigo Araujo"]
```
Comment (vnepveu) [inline fuels-abigen-macro/tests/test_projects/two-structs/Cargo.toml:4]: Author and licence I think (#49)

**Review by vnepveu — CHANGES_REQUESTED** [review]
Nitpicking on author/licence

File: fuels-abigen-macro/tests/test_projects/two-structs/Forc.toml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+[project]
+name = "demo"
+author = "Rodrigo Araujo"
```
[the remark this answers is not in the document]
Comment (digorithm) [reply fuels-abigen-macro/tests/test_projects/two-structs/Forc.toml:3]: Yup nice catch!

**Review by vnepveu — APPROVED** [review]
LGTM

**Pull request merged**